### DISCUSSION TITLE
Q: 是否有 openapi 的包提供出来 || Q: Is there any openapi package provided?

### DISCUSSION THREAD

**Discussion by iambinlin** [walinejs/waline#2194] [Q&A] [answered]
## 背景

自定义博客开发，希望仅用到访问量统计，在不引入 `UI` 包的前提下，期望可以拥有 `openapi` ，直接在代码中调用设置状态。

直白来讲就是跟 `UI` 无关，不做强绑定，现在包的形式看起来有些慢。

### Website URL | 问题网站

https://blog.binlin.wang/posts/goodbye-clean-code.20231211

### Where your waline deploy? | 服务部署在哪里？

Vercel (Default)

### Where your comment data store? | 数据存储在哪里？

LeanCloud(https://leancloud.app)

---
## background

For custom blog development, I hope to only use traffic statistics. Without introducing the `UI` package, I hope to have `openapi` and directly call the setting status in the code.

To put it bluntly, it has nothing to do with `UI`, and without strong binding, the current package format seems a bit slow.

### Website URL | Problem website

https://blog.binlin.wang/posts/goodbye-clean-code.20231211

### Where your waline deploy? | Where is the service deployed?

Vercel (Default)

### Where your comment data store? | Where is the data stored?

LeanCloud(https://leancloud.app)

**Comment by lizheming** [comment, marked as answer]
https://www.npmjs.com/package/@waline/api
https://github.com/walinejs/waline/tree/main/packages/api 

具体的 API 得你看看代码了，暂时还没有文档


---
https://www.npmjs.com/package/@waline/api
https://github.com/walinejs/waline/tree/main/packages/api

You have to look at the code for the specific API. There is no documentation yet.

**Reply by iambinlin** [reply]
好的，谢谢，我来封装一下。

---
Okay, thank you, I'll encapsulate it.
